### PR TITLE
Make the TxTable Tag column fixed (instead of advice)

### LINKED ISSUE #897: Make the TxTable Tag column fixed (instead of advice)
See https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/764#issuecomment-1310185413 for reference

### PR DESCRIPTION
**Pull request by xiaodino**
https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/897

### CODE REVIEW

File: zkevm-circuits/src/table.rs [hunk 1]
```diff
@@ -153,14 +152,30 @@ impl TxTable {
...
+                            self.index,
+                            offset,
+                            || Value::known(row[2]),
+                        )?;
+                        region.assign_advice(
+                            || format!("tx table row {}", offset),
+                            self.value,
+                            offset,
+                            || Value::known(row[3]),
+                        )?;
```
Comment (Brechtpd) [inline zkevm-circuits/src/table.rs:178]: Perhaps still nicer to use a for loop here for all the advice columns.

File: zkevm-circuits/src/table.rs [hunk 2]
```diff
@@ -170,9 +185,20 @@ impl TxTable {
...
 }
 
-impl DynamicTableColumns for TxTable {
-    fn columns(&self) -> Vec<Column<Advice>> {
-        vec![self.tx_id, self.tag, self.index, self.value]
+impl TxTable {
+    pub(crate) fn columns(&self) -> Vec<Column<Advice>> {
+        vec![self.tx_id, self.index, self.value]
+    }
+}
```
[the remark this answers is not in the document]
Comment (xiaodino) [reply zkevm-circuits/src/table.rs]: Removed.

File: zkevm-circuits/src/table.rs (hunk 1 above)
Comment (xiaodino) [reply zkevm-circuits/src/table.rs]: Updated. Use a for loop for all the advice columns.

**Comment by xiaodino** [comment]
@ed255 Could you take a look?

**Review by ed255 — APPROVED** [review]
LGTM!

**Review by Brechtpd — APPROVED** [review]

**Comment by ed255** [comment]
@xiaodino can you rebase the branch of this PR with main?  Then we can merge :)

**Pull request merged**